commit 4acf1de35f41549e60c3c02a8defa7cb95eabdf2
Author: Kees Cook <kees@kernel.org>
Date:   Fri Jan 6 19:47:05 2023 -0800

    kunit: memcpy: Split slow memcpy tests into MEMCPY_SLOW_KUNIT_TEST
    
    Since the long memcpy tests may stall a system for tens of seconds
    in virtualized architecture environments, split those tests off under
    CONFIG_MEMCPY_SLOW_KUNIT_TEST so they can be separately disabled.
    
    Reported-by: Guenter Roeck <linux@roeck-us.net>
    Link: https://lore.kernel.org/lkml/20221226195206.GA2626419@roeck-us.net
    Reviewed-by: Nick Desaulniers <ndesaulniers@google.com>
    Reviewed-and-tested-by: Guenter Roeck <linux@roeck-us.net>
    Reviewed-by: David Gow <davidgow@google.com>
    Cc: Andrew Morton <akpm@linux-foundation.org>
    Cc: Nathan Chancellor <nathan@kernel.org>
    Cc: linux-hardening@vger.kernel.org
    Signed-off-by: Kees Cook <keescook@chromium.org>

diff --git a/lib/memcpy_kunit.c b/lib/memcpy_kunit.c
index 89128551448d..887926f04731 100644
--- a/lib/memcpy_kunit.c
+++ b/lib/memcpy_kunit.c
@@ -309,6 +309,8 @@ static void set_random_nonzero(struct kunit *test, u8 *byte)
 
 static void init_large(struct kunit *test)
 {
+	if (!IS_ENABLED(CONFIG_MEMCPY_SLOW_KUNIT_TEST))
+		kunit_skip(test, "Slow test skipped. Enable with CONFIG_MEMCPY_SLOW_KUNIT_TEST=y");
 
 	/* Get many bit patterns. */
 	get_random_bytes(large_src, ARRAY_SIZE(large_src));